sympy/discrete/recurrence.py

```diff
@@ -0,0 +1,132 @@
+"""
+Recurrence
+"""
+from __future__ import print_function, division
+
+from sympy.core import S, Symbol, sympify
+from sympy.core.compatibility import as_int, range, iterable
+
+def linrec(coeffs, init, n):
+    r"""
+    Evaluation of univariate linear recurrences of homogeneous type
+    having coefficients independent of the recurrence variable.
+
+    Parameters
+    ==========
+
+    coeffs : iterable
+        coefficients of recurrence
+    init : iterable
+        initial values of recurrence
+    n : Integer
+        point of evaluation of recurrence
+
+    Explanation
```

Comment:
The name of this section should probably be `Notes`. See https://numpydoc.readthedocs.io/en/latest/format.html#sections.